tests/tests_integration/test_api/test_workflows.py

```diff
@@ -0,0 +1,348 @@
+def workflow_list(cognite_client: CogniteClient) -> WorkflowList:
+    workflow1 = WorkflowCreate(
+        external_id="integration_test-workflow1",
+        description="This is  workflow for testing purposes",
+    )
+    workflow2 = WorkflowCreate(
+        external_id="integration_test-workflow2",
+        description="This is  workflow for testing purposes",
+    )
+    workflows = [workflow1, workflow2]
+    listed = cognite_client.workflows.list()
+    existing = {w.external_id for w in listed}
+    call_list = False
+    for workflow in workflows:
+        if workflow.external_id not in existing:
+            call_list = True
+            cognite_client.workflows.upsert(workflow)
+    if call_list:
+        return cognite_client.workflows.list()
+    return listed
+
+
+@pytest.fixture
+def workflow_version_list(cognite_client: CogniteClient) -> WorkflowVersionList:
+    workflow_id = "integration_test-workflow_with_versions"
+    version1 = WorkflowVersionCreate(
+        workflow_external_id=workflow_id,
+        version="1",
+        workflow_definition=WorkflowDefinitionCreate(
+            tasks=[
+                WorkflowTask(
+                    external_id=f"{workflow_id}-1-task1",
+                    parameters=TransformationTaskParameters(
+                        external_id="None-existing-transformation",
+                    ),
+                )
+            ],
+        ),
+    )
+    version2 = WorkflowVersionCreate(
+        workflow_external_id=workflow_id,
+        version="2",
+        workflow_definition=WorkflowDefinitionCreate(
+            tasks=[
+                WorkflowTask(
+                    external_id=f"{workflow_id}-2-task1",
+                    parameters=CDFTaskParameters(
+                        resource_path="/dummy/no/real/resource/path",
+                        method="GET",
+                        body={"limit": 1},
+                    ),
+                )
+            ],
+        ),
+    )
+    listed = cognite_client.workflows.versions.list(workflow_ids=workflow_id)
+    existing = {w.version for w in listed}
+    call_list = False
+    for version in [version1, version2]:
+        if version.version not in existing:
+            call_list = True
+            cognite_client.workflows.versions.upsert(version)
+    if call_list:
+        return cognite_client.workflows.versions.list(workflow_ids=workflow_id)
+    return listed
+
+
+@pytest.fixture
+def cdf_function_add(cognite_client: CogniteClient) -> Function:
+    external_id = "integration_test-workflow-cdf_function_add"
+    add_function = cognite_client.functions.retrieve(external_id=external_id)
+    if add_function is not None:
+        return add_function
+
+    def handle(client, data: dict):
+        output = data.copy()
+        output["sum"] = output["a"] + output["b"]
+        return output
+
+    deployed = cognite_client.functions.create(name="Add", external_id=external_id, function_handle=handle)
+    return deployed
```

Comment:
I see we have our first functions-integration-test-function, nice. I'll add a task on adding some integration tests (call, get_logs, get_calls, schedules etc.)

Should this be marked with `scope="session"`?